script.js

```diff
@@ -1,4 +1,565 @@
+  if (stageOfGame == WAITING_FOR_PLAYER_TO_CLICK_SUBMIT) {
+    myOutputValue = `The deck is shuffled. Enter "deal" to deal cards.`;
+
+    stageOfGame = PLAYER_TO_ENTER_DEAL;
+  }
+
+  // Player must enter "deal" to deal cards
+  else if (input == `deal` && stageOfGame == PLAYER_TO_ENTER_DEAL) {
+    stageOfGame = DEAL_FIRST_2_CARDS;
+
+    // Dealer draws 2 cards
+    var dealersCard1 = shuffledDeck.pop();
+    console.log(`Dealer's card: ${dealersCard1.name} of ${dealersCard1.suit}`);
+    var dealersCard2 = shuffledDeck.pop();
+    console.log(`Dealer's card: ${dealersCard2.name} of ${dealersCard2.suit}`);
+
+    // Change all of dealer's picture cards (if any) to = 10
+    if (
+      dealersCard1.name == "💂‍♂️ Jack" ||
+      dealersCard1.name == "👸 Queen" ||
+      dealersCard1.name == "🤴 King"
+    ) {
+      dealersCard1.rank = 10;
+    }
+
+    if (
+      dealersCard2.name == "💂‍♂️ Jack" ||
+      dealersCard2.name == "👸 Queen" ||
+      dealersCard2.name == "🤴 King"
+    ) {
+      dealersCard2.rank = 10;
+    }
+
+    // Dealer's total score from first 2 cards
+    dealersCard1PlusCard2 = dealersCard1.rank + dealersCard2.rank;
+    console.log(`Dealer's cards 1 and 2 Total = ${dealersCard1PlusCard2}`);
+
+    // Push Dealer's 1st and 2nd card into array
+    arrayOfDealersCards.push(`${dealersCard1.name} of ${dealersCard1.suit}`);
+    arrayOfDealersCards.push(` ${dealersCard2.name} of ${dealersCard2.suit}`);
+    console.log(arrayOfDealersCards);
+
+    // Player draws 2 cards
+    var playerCard1 = shuffledDeck.pop();
+    console.log(`Player card: ${playerCard1.name} of ${playerCard1.suit}`);
+    var playerCard2 = shuffledDeck.pop();
+    console.log(`Player card: ${playerCard2.name} of ${playerCard2.suit}`);
+
+    // Change all of player's picture cards (if any) to = 10
+    if (
+      playerCard1.name == "💂‍♂️ Jack" ||
+      playerCard1.name == "👸 Queen" ||
+      playerCard1.name == "🤴 King"
+    ) {
+      playerCard1.rank = 10;
+    }
+
+    if (
+      playerCard2.name == "💂‍♂️ Jack" ||
+      playerCard2.name == "👸 Queen" ||
+      playerCard2.name == "🤴 King"
+    ) {
+      playerCard2.rank = 10;
+    }
+
+    // Player's total score from first 2 cards
+    playerCard1PlusCard2 = playerCard1.rank + playerCard2.rank;
+    console.log(`Player's cards 1 and 2 Total = ${playerCard1PlusCard2}`);
+
+    // Push Player's 1st and 2nd card into array
+    arrayOfPlayerCards.push(`${playerCard1.name} of ${playerCard1.suit}`);
+    arrayOfPlayerCards.push(` ${playerCard2.name} of ${playerCard2.suit}`);
+    console.log(arrayOfPlayerCards);
+
+    // If player gets a 10 + an Ace = Blackjack!; Player wins; Dealer loses
+
+    if (
+      (playerCard1.rank == 1 && playerCard2.rank == 10) ||
+      (playerCard1.rank == 10 && playerCard2.rank == 1)
+    ) {
+      console.log(`player blackjack!`);
+      myOutputValue = `You drew:<br>${arrayOfPlayerCards}.<br>Your total score now is 21.<br><br>Blackjack! You win!`;
```

Comment:
Imo you can just immediately push cards from the deck to the player and dealer's hands from line 53 onwards and do a for loop to loop through each card to give the card its rank.